bench.zig

```diff
@@ -35,7 +35,7 @@ pub fn benchmark(comptime B: type) !void {
         var res = [_]u64{ 0, 0, 0 };
         res = try printBenchmark(writer, res, "Benchmark", "", "Iterations", "Mean(ns)");
         inline for (functions) |f| {
-            for (args) |_, i| {
+            inline for (args) |_, i| {
```

Comment:
I wonna avoid inline loops where possible. We can change this code to use a `while` loop, and then we don't need to inline it:
```
var i: usize = 0;
while (i < args.len) : (i += 1) {
```

Reply:
Right, that one didn't need to be inlined.

Maybe I can get around the need for comptime args all together with some clever wrapping.

Reply:
I think your testcase shows a good benchmark for why this patch is needed, so I wonna merge it. One problem with this is, that this now doesn't allow `args` to be generated during runtime, before the benchmarks are run. Currently this can be done like this:
```zig
test "benchmark generics" {
    const B = struct {
        pub var args: []const []const u8 = undefined;

        pub fn func(a: []const u8) T {}
    };
    B.args = &[_][]const u8{"yo"};

    try benchmark(B);
}
```

I'll think over this for a bit, but currently I'm leaning on just breaking this usecase. Any thoughts are welcome.

Reply:
All I can come up with is duplicating the inner loop of `benchmark()`, and selecting one or the other with a comptime boolean. 

```zig
if (comptime_args) {
    inline for (args) |arg, index| {
        ...
    }
} else {
    for (args) |arg, index| {
        ...
    }
}
```

Not very nice and DRY at all. Maybe wrapping all local state in a struct with a separate function for the inner block could work.

Reply:
Hmm. Yea that does seem like the only option doesn't it. Well, for now I think we just merge this and then figure this out when someone needs it.